### PR TITLE
FIX-#6585: avoid `FutureWarning`s in `rolling` unless necessary

### LINKED ISSUE #6585: Using `.rolling` func leads to getting `FutureWarning: The 'axis' keyword in DataFrame.rolling is deprecated and will be removed in a future version. Call the method without the axis keyword instead.`

### PR DESCRIPTION
**Pull request by anmyachev**
## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #6585 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

File: modin/core/storage_formats/pandas/query_compiler.py [hunk 1]
```diff
@@ -1519,9 +1519,9 @@ def expanding_corr(
         )
     )
     rolling_quantile = Fold.register(
-        lambda df, rolling_kwargs, quantile, interpolation, **kwargs: pandas.DataFrame(
+        lambda df, rolling_kwargs, q, interpolation, **kwargs: pandas.DataFrame(
             df.rolling(**rolling_kwargs).quantile(
-                quantile=quantile, interpolation=interpolation, **kwargs
```
Comment (anmyachev) [inline modin/core/storage_formats/pandas/query_compiler.py:1524]: Deprecated kwarg name.

File: modin/pandas/test/conftest.py [hunk 2]
```diff
@@ -24,8 +24,8 @@ def pytest_collection_modifyitems(items):
         ):
             for item in items:
                 if item.name in (
-                    "test_dataframe_dt_index[3s-both-DateCol-0]",
-                    "test_dataframe_dt_index[3s-right-DateCol-0]",
+                    "test_dataframe_dt_index[3s-both-DateCol-_NoDefault.no_default]",
```
Comment (anmyachev) [inline modin/pandas/test/conftest.py:27]: To be align with the new value of axis for `test_dataframe_dt_index` test.

File: modin/pandas/test/test_rolling.py [hunk 3]
```diff
@@ -34,7 +35,19 @@
 # have too many such instances.
 # TODO(https://github.com/modin-project/modin/issues/3655): catch all instances
 # of defaulting to pandas.
-pytestmark = pytest.mark.filterwarnings(default_to_pandas_ignore_string)
+pytestmark = [
```
Comment (anmyachev) [inline modin/pandas/test/test_rolling.py:38]: The same way as it done for `test_groupby.py`.

File: modin/pandas/test/test_rolling.py [hunk 4]
```diff
@@ -34,7 +35,19 @@
...
+    pytest.mark.filterwarnings(default_to_pandas_ignore_string),
+    # TO MAKE SURE ALL FUTUREWARNINGS ARE CONSIDERED
+    pytest.mark.filterwarnings("error::FutureWarning"),
+    # IGNORE FUTUREWARNINGS MARKS TO CLEANUP OUTPUT
+    pytest.mark.filterwarnings(
+        "ignore:Support for axis=1 in DataFrame.rolling is deprecated:FutureWarning"
+    ),
+    # FIXME: these cases inconsistent between modin and pandas
```
Comment (dchigarev) [inline modin/pandas/test/test_rolling.py:46]: in which way they're inconsistent, do we have an issue for that?

**Review by dchigarev — APPROVED** [review]

File: modin/pandas/test/test_rolling.py (hunk 4 above)
Comment (anmyachev) [reply modin/pandas/test/test_rolling.py:46]: Modin does not show this warning in cases where Pandas does. Not yet.

Comment (anmyachev) [reply modin/pandas/test/test_rolling.py:46]: @dchigarev let's merge, I will create issue later.

**Pull request merged**